src/commands/images/runImage.ts

```diff
@@ -19,21 +20,21 @@ async function runImageCore(context: IActionContext, node: ImageTreeItem | undef
     if (!node) {
         node = await ext.imagesTree.showTreeItemPicker<ImageTreeItem>(ImageTreeItem.contextValue, {
             ...context,
-            noItemFoundErrorMessage: 'No images are availalbe to run'
+            noItemFoundErrorMessage: 'No images are available to run'
         });
     }
 
     const inspectInfo = await node.getImage().inspect();
-    const ports: string[] = inspectInfo.Config.ExposedPorts ? Object.keys(inspectInfo.Config.ExposedPorts) : [];
 
-    let options = `--rm ${interactive ? '-it' : '-d'}`;
-    if (ports.length) {
-        const portMappings = ports.map((port) => `-p ${port.split("/")[0]}:${port}`); // 'port' is of the form number/protocol, eg. 8080/udp.
-        // In the command, the host port has just the number (mentioned in the EXPOSE step), while the destination port can specify the protocol too
-        options += ` ${portMappings.join(' ')}`;
-    }
+    const terminalCommand = await selectRunCommand(
+        context,
+        [node.fullTag],
```

Comment:
The contexts are part of the customization contract, so I think it makes sense to simply make them a required argument to the `selectXxxCommand()` function.  (This is especially easy if the context is already a required argument, such as here.)